common/djangoapps/student/views/management.py

```diff
@@ -912,6 +912,28 @@ def validate_new_email(user, new_email):
         raise ValueError(_('Old email is the same as the new email.'))
 
 
+def validate_secondary_email(user_profile, new_email):
+    """
+    Enforce valid email addresses.
+    """
+    from openedx.core.djangoapps.user_api.accounts.api import get_email_validation_error, \
+        get_email_existence_validation_error, get_secondary_email_validation_error
+
+    if get_email_validation_error(new_email):
+        raise ValueError(_('Valid e-mail address required.'))
+
+    if user_profile.secondary_email and new_email == user_profile.secondary_email:
+        raise ValueError(_('Old email is the same as the new email.'))
+
+    message = get_email_existence_validation_error(new_email)
```

Comment:
please add a comment mentioning that we need unique secondary email across primary user emails and secondary user emails.